dag-cbor/src/error.rs

```diff
@@ -115,3 +115,8 @@ pub struct UnexpectedEof;
 #[derive(Debug, Error)]
 #[error("Invalid Cid prefix: {0}")]
 pub struct InvalidCidPrefix(pub u8);
+
+/// A duplicate key within a map.
+#[derive(Debug, Error)]
+#[error("Duplicate map key.")]
+pub struct DuplicateKey;
```

Comment:
would be nice to inform which key is duplicate

Reply:
Yes, it would be nice, but it's a pain to implement, hence I didn't do it.

Reply:
I should have written a more elaborate reply: The problem is that it's generic over `K`. So I'd need to add the `Display` trait. That would then bleed into other things. I found it a nice to have, but not important enough to figure out all the details.